### DISCUSSION TITLE
Does fooocus support control net?

### DISCUSSION THREAD

**Discussion by martinenkoEduard** [lllyasviel/Fooocus#3572] [General]
Does fooocus support control net?

**Comment by Crocodilian12** [comment]
That's what the image prompts are, ControlNets. The basic image prompt is a tweaked version of IP Adaptor; PyraCanny and CPDS are both ControlNets . . . I'm not sure about the Faceswap tools.

**Reply by ckotty-alex** [reply]
> That's what the image prompts are, ControlNets. The basic image prompt is a tweaked version of IP Adaptor; PyraCanny and CPDS are both ControlNets . . . I'm not sure about the Faceswap tools.

faceswap is  faceid ipadapter

**Comment by ckotty-alex** [comment]
by the way, openpose works too in Fooocus. I've changed control-lora-canny to controlnet union, turn on "skip preprocessors" , then insert openpose images to image promp "canny" and it's really works

**Reply by Yogesh-DevHub** [reply]
Thanks for mentioning, was using comfy UI for that. Will try in fooocus.

**Comment by cubiwan** [comment]
I have tested using openpose images and it works, no code changes are necessary:

Download diffusion_pytorch_model.safetensors

Save in controlnet folder

Rename to control-lora-canny-rank128.safetensors

Launch Fooocus

Advanced tab, Control tab, check Skip Preprocessors

Image Prompt tab with an openpose skeleton image (512x512 works well)

Select PyraCanny option

Select Aspect Ratio 1:1 (with another the generated image appears cut off)

Write a prompt and generate image.

**Reply by briziomusic** [reply]
> I have tested using openpose images and it works, no code changes are necessary:
> 
> Download diffusion_pytorch_model.safetensors
> 
> Save in controlnet folder
> 
> Rename to control-lora-canny-rank128.safetensors
> 
> Launch Fooocus
> 
> Advanced tab, Control tab, check Skip Preprocessors
> 
> Image Prompt tab with an openpose skeleton image (512x512 works well)
> 
> Select PyraCanny option
> 
> Select Aspect Ratio 1:1 (with another the generated image appears cut off)
> 
> Write a prompt and generate image.

Sorry, could you please guide me? I'm not able to reproduce.

is this image uploaded to 'pyracanny' okay (512x512)?
![test1](https://github.com/user-attachments/assets/0196c57b-f761-416c-abda-ec6e3dba691a)

result

![test2](https://github.com/user-attachments/assets/47fb75ee-a7fb-47a1-8945-599aef48a077)

controlnet folder

![test3](https://github.com/user-attachments/assets/29941d70-3e93-4d3c-aaa9-ac0694301166)

the file was downloaded from 
https://huggingface.co/InstantX/InstantID/blob/main/ControlNetModel/diffusion_pytorch_model.safetensors
and renamed to control-lora-canny-rank128.safetensors

what am I doing wrong?

**Reply by Yogesh-DevHub** [reply]
Get this one https://huggingface.co/xinsir/controlnet-union-sdxl-1.0/tree/main

it will work on default stop-at and weight, try keeping aspect ratio same as the uploaded image, it has an impact on result

**Reply by briziomusic** [reply]
> Get this one https://huggingface.co/xinsir/controlnet-union-sdxl-1.0/tree/main
> 
> it will work on default stop-at and weight, try keeping aspect ratio same as the uploaded image, it has an impact on result

Thank you. It's working!

![test4](https://github.com/user-attachments/assets/bdde121f-85b1-4a3c-a78a-bad58cdf4ea9)
